src/blog/2024/04/using-mongodb-with-node-red.md

```diff
@@ -0,0 +1,281 @@
+
+4. Click on the MongoDB4 node that we have added while configuring the MongoDB4 node.
+5. Select the connection which we have added, and enter the collection name as “customers”. The collection will be created if it does not exist already in your MongoDB database.
+6. Set the operation as “insertOne” and keep the other things unchanged.
+
+!["Screenshot displaying configuration of MongoDB 4 node for inserting data"](./images/using-mongodb-with-node-red-mongodb-insertone-node-configuration.png "Screenshot displaying configuration of MongoDB 4 node for inserting data"){data-zoomable}
+
+7 Connect the nodes' wires as shown in the below image.
+
+!["Screenshot displaying connections of wires in the 'Insert Data into Database' flow"](./images/using-mongodb-with-node-red-mongodb-insertone-flow.png "Screenshot displaying connections of wires in the 'Insert Data into Database' flow"){data-zoomable}
+
+### Retrieving Customer Data from Database
+
+- Drag the inject node onto the canvas. In the inject node, set `msg.payload` as an empty object. Additionally, set the inject node to send the payload after a specific interval of time to update the data in the table.
+- Drag the MongoDB4 node onto canvas, make sure you have selected the connection, and enter “find” into the operation input field.
+
+!["Screenshot displaying configuration of MongoDB 4 node for retrieving data"](./images/using-mongodb-with-node-red-mongodb-find-node-configuration.png "Screenshot displaying configuration of MongoDB 4 node for retrieving data"){data-zoomable}
+
+- Drag the ui-table widget onto the canvas, create a new ui-group for it, and set max rows according to your preference.
+
+!["Screenshot displaying ui-table widget configuration"](./images/using-mongodb-with-node-red-table-widget.png "Screenshot displaying ui-table widget configuration"){data-zoomable}
+
+- Connect the nodes' wires as shown in the below image.
+
+!["Screenshot displaying connections of wires in the 'Retrive Data from Database' flow"](./images/using-mongodb-with-node-red-mongodb-find-flow.png "Annotomy of MongoDB document"){data-zoomable}
+
+### Updating Customer Data to Database
+
+- Drag the ui-form widget onto the canvas, create a new ui-group for it, and add form elements for "id" and "status".
+
+!["Screenshot displaying form widget configuration to update data in MongoDB"](./images/using-mongodb-with-node-red-insert-data-form.png "Screenshot displaying form widget configuration to update data in MongoDB"]{data-zoomable}
+
+- Drag the change node onto the canvas. In the change node, set `msg.payload` as the following JSON object:
+
+```json
+[
+    // Specifying the criteria to identify the document to update, based on its _id field
```

Comment:
Not sure if we should allow people to copy+paste illegal JSON?

Reply:
How about if we add the following note before code blocks:
Note:- *Please note that the comments provided are for explanation purposes only and should not be copied along with the code*

Reply:
Not as a comment in the code I think, just above it?